app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java

```diff
@@ -4,14 +4,65 @@
 
 public class CalculatorModel implements CalculatorContract.CalculatorModelContract {
 
-    private String firstOperand;
+    private String firstOperand = "";
+    private String secondOperand = "";
+    private String operator = "";
+    private String result = "";
+
 
     @Override
     public void saveNumber(String number) {
-        firstOperand = number;
+        if (operator.isEmpty()) {
+            firstOperand = firstOperand + number;
+        } else {
+            secondOperand = secondOperand + number;
+        }
+    }
+
+    public String getPartialResult() {
+
+        return (firstOperand + operator + secondOperand);
+    }
+
+    @Override
+    public String getFullResult() {
+
+        switch (operator) {
+
+            case "+":
```

Comment:
no hardcode, create a constant for all the operations pls

Reply:
Done. Thx